packages/xo-web/src/xo-app/logs/backup-ng/log-alert-body.js

```diff
@@ -72,9 +138,199 @@ const Warnings = ({ warnings }) =>
+          _('taskTransferredDataSpeed'),
+          formatSpeed(task.transfer.size, task.transfer.duration)
+        )}
+      </div>
+    )}
+    {task.merge !== undefined && (
+      <div>
+        {_.keyValue(_('taskMergedDataSize'), formatSize(task.merge.size))}
+        <br />
+        {_.keyValue(
+          _('taskMergedDataSpeed'),
+          formatSpeed(task.merge.size, task.merge.duration)
+        )}
+      </div>
+    )}
+    {task.isFull !== undefined &&
+      _.keyValue(_('exportType'), task.isFull ? 'full' : 'delta')}
+  </div>
+)
+
+const PoolTask = ({ children, task }) => (
+  <div>
+    <Pool id={task.data.id} link newTab />{' '}
+    <TaskStateInfos status={task.status} />
+    <Warnings warnings={task.warnings} />
+    {children}
+    <TaskStart task={task} />
+    <TaskEnd task={task} />
+    <TaskDuration task={task} />
+    <TaskError task={task} />
+  </div>
+)
+
+const XoTask = ({ children, task }) => (
+  <div>
+    <Icon icon='menu-xoa' /> XO <TaskStateInfos status={task.status} />
+    <Warnings warnings={task.warnings} />
+    {children}
+    <TaskStart task={task} />
+    <TaskEnd task={task} />
+    <TaskDuration task={task} />
+    <TaskError task={task} />
+  </div>
+)
+
+const SnapshotTask = ({ task }) => (
+  <div>
+    <Icon icon='task' /> {_('snapshotVmLabel')}{' '}
+    <TaskStateInfos status={task.status} />
+    <Warnings warnings={task.warnings} />
+    <TaskStart task={task} />
+    <TaskEnd task={task} />
+    <TaskError task={task} />
+  </div>
+)
+
+const RemoteTask = ({ children, task }) => (
+  <div>
+    <Remote id={task.data.id} link newTab />{' '}
+    <TaskStateInfos status={task.status} />
+    <Warnings warnings={task.warnings} />
+    {children}
+    <TaskStart task={task} />
+    <TaskEnd task={task} />
+    <TaskDuration task={task} />
+    <TaskError task={task} />
+  </div>
+)
+
+const SrTask = ({ children, task }) => (
+  <div>
+    <Sr id={task.data.id} link newTab /> <TaskStateInfos status={task.status} />
+    <Warnings warnings={task.warnings} />
+    {children}
+    <TaskStart task={task} />
+    <TaskEnd task={task} />
+    <TaskDuration task={task} />
+    <TaskError task={task} />
+  </div>
+)
+
+const TransferMergeTask = ({ task }) => {
+  const size = get(() => task.result.size)
+  return (
+    <div>
+      <Icon icon='task' /> {task.message}{' '}
+      <TaskStateInfos status={task.status} />
+      <Warnings warnings={task.warnings} />
+      <TaskStart task={task} />
+      <TaskEnd task={task} />
+      <TaskDuration task={task} />
+      <TaskError task={task} />
+      {size > 0 && (
+        <div>
+          {_.keyValue(_('operationSize'), formatSize(size))}
+          <br />
+          {_.keyValue(
+            _('operationSpeed'),
+            formatSpeed(size, task.end - task.start)
+          )}
+        </div>
+      )}
+    </div>
+  )
+}
+
+const TaskLi = ({ className, task, ...props }) => {
+  const type = get(() => task.data.type)
+
+  if (type === 'VM') {
```

Comment:
IMHO, better be safe if it don't cost. If we add a task which it isn't handled by our implementation it will break the UI.

Reply:
https://github.com/vatesfr/xen-orchestra/pull/4014#discussion_r272160458 :thinking:

Reply:
I remove `if(type !== undefined)`

Reply:
Ok then I still don't get what's wrong with [this code](https://github.com/vatesfr/xen-orchestra/pull/4014#discussion_r267704696).

Reply:
Empty `<li />` when the `type` is not handled by our implementation ?

Reply:
Ok so it doesn't *break* the UI. Plus it should never happen, right? So in the most unexpected case, it would simply leave an empty `li`. Sounds like a pretty good behaviour to me.

Reply:
I'll suggest a declarative implementation.